CIP-1694/README.md

```diff
@@ -0,0 +1,664 @@
+| 6(b). Treasury withdrawal, $[T_1, T_2)$              | :heavy_check_mark:       | $P_{6b}$ | $Q_{6b}$ | SPO Vote      | $R_{6b}$ |
+| 6(c). Treasury withdrawal, $[T_2, T_3)$              | :heavy_check_mark:       | $P_{6c}$ | $Q_{6c}$ | SPO Vote      | $R_{6c}$ |
+
+Some of the parameters given in this table ( $P_1$ ... $R_{6c}$, $T_1$ ... $T_3$ ) may be updatable protocol parameters, but others should be hard-coded. This proposal deliberately leaves both this choice and the choice of actual parameter values open for discussion.
+
+> **Note**
+> For all treasury withdrawals, the withdrawal threshold is the **total** amount of Lovelace that is withdrawn by the action, not the amount of any single withdrawal if the action specifies more than one withdrawal.
+
+##### Governance Action Lifecycle
+
+Governance actions are checked for ratification only on an epoch boundary. This delay allows everyone to vote on each proposal and prove that they are active.
+
+**At most one** governance action **of each type** (i.e. one from each of the six categories listed above) can be staged for enactment in any given epoch. For example, there can only be one treasury withdrawal action in a single epoch (it may, however, comprise many individual withdrawals).
+
+Once ratified, actions will be staged for enactment. Actions that have been staged will be enacted on the following epoch boundary, unless they are dropped. All submitted governance actions will therefore either:
+
+1. be **ratified**;
+2. be **dropped** as a result of some higher priority action; or else
+3. will **expire** after a number of epochs.
+
+Deposits are returned immediately when either:
+
+1. a ratified action is **enacted**;
+1. the action **expires**; or
+1. a ratified action is **dropped**.
+
+Ratification is explained in more detail later in this document.
+
+>**Note:**
+>This design means that the first governance action of a given type to be ratified will be staged for enactment in an epoch.
+
+#### Enactment
+
+Ratified actions may be enacted at an epoch boundary. During enactment, actions in the staging group for the current epoch are prioritized as follows:
+
+1. Motion of no-confidence;
+2. New Constitutional Committee or a change to the quorum size;
+3. Updates to the Constitution;
+4. Hard Fork initiation;
+5. Protocol parameter changes;
+6. Treasury withdrawals.
+
+As described above, at most one action of each type may be enacted in any given epoch.
+So a ratified _"Motion of no-confidence"_ action will be enacted first if there is one, and so on.
+
+A successful _"Motion of no-confidence"_, the election of a new Constitutional Committee, or a constitutional change invalidates *all* other **unenacted** governance actions (whether or not they have been ratified),
+_causing them to be immediately **dropped** without ever being enacted_. Deposits for dropped actions will be returned immediately.
```

Comment:
One of the ways that we prevent someone from spamming the list of governance actions with bogus entries is by requiring a deposit, and holding on to the deposit for a nontrivial amount of time. I suppose that an action could be removed at the user's wish, this wouldn't cause the queue to fill up with noise, but we did want some incentives to remove the non-serious actions.

Reply:
There are two problems with this:

First, there's the question of whether the submitter should have this power. It would essentially give anyone a veto power on their own proposals. Now, imagine someone being against a change that is popular. Then that person could attack this mechanism by making the proposal, waiting until it gets close to passing and then withdraw it again. Now that proposal is effectively delayed by some time and everyone lost some fees. This means that you'd now have to trust that the submitter of a proposal is not malicious if you want to avoid that scenario.

Second, it's also a bit complicated to figure out good semantics. How do we authenticate manually dropping a GA? We could give transactions signed by the same set of keys as the submitting Tx this power, but this becomes weird in the presence of scripts. Alternatively, we could add a credential to the submission and then we'd have to check for that. We'd also need to add this feature to the transaction body and add logic for it.

Maybe there are good ways of dealing with these issues and good reasons for adding this feature, but I think this is best left for a future CIP/CPS.

Reply:
Thank you for your insights @JaredCorduan & @WhatisRT.

> Now, imagine someone being against a change that is popular. Then that person could attack this mechanism by making the proposal, waiting until it gets close to passing and then withdraw it again. Now that proposal is effectively delayed by some time and everyone lost some fees.

The obvious answer is that if a proposer abuses this mechanism, they would lose their deposit, or the deposit is used to compensate the fees paid by the voters. The tricky part is who would decide if a GA was dropped "correctly" or not? At the moment, I cannot think of an exemplary process for this.

I would also like to understand the thoughts behind updateability of GAs. While I understand why we can't just have the proposer update a GA moments before it would pass, I believe it should be necessary to update GA and, maybe similar to my suggestion above, use the deposit to compensate fees that were paid by voters that already voted on this GA.

Reply:
> The tricky part is who would decide if a GA was dropped "correctly" or not?

That, and how to make sure that process can't get attacked. This sounds like it might get quite complicated to get this right, for not too much gain.

Updating is arguably the same as withdrawing an action and submitting a new one, suffering from the same problems. For now, if a GA needs to be amended, the correct path would just be to make a new one and convince the community that they should vote for the new one and change their vote on the old one to `No`. What's nice about this is that everyone has the power to do that.

Reply:
I worry that allowing updating would make things more complex and confusing. At the very least, we would need to remove all the votes from the goverance action and change the timelines. Which, at that point, the only thing you get by allowing updates is that the proposer doesn't need a second deposit while they wait.